src/topics/resolver/global-resolver.ts

```diff
@@ -25,56 +36,69 @@ export class GlobalResolver {
     this.ignoreAccountErrorsWhenResolving =
       ignoreAccountErrorsWhenResolving === "true";
 
-    const factory = regExps.includes("^test:")
+    this.factory = regExps.includes("^test:")
       ? testResolverFactory
       : resolverFactory;
 
     regExps.map((regexp: string) => {
-      if (factory[regexp] === undefined) {
+      if (this.factory[regexp] === undefined) {
         throw new Error(`The RegExp ${regexp} is not mapped to any resolver`);
       }
       this.resolverRouter.push({
-        resolver: factory[regexp].resolver,
+        resolver: this.factory[regexp].resolver,
         regExp: new RegExp(regexp),
-        accountType: factory[regexp].accountType,
+        accountSource: this.factory[regexp].accountSource,
+        accountType: this.factory[regexp].accountType,
       });
     });
   }
 
   public async resolveAll(rawData: FetchedData): Promise<ResolveAllType> {
     const updatedRawData: FetchedData = rawData;
-    const resolvedIdentifierData: FetchedData = {};
-    const accountTypes: AccountSource[] = [];
+    const accountSources: AccountSource[] = [];
+    let resolvedIdentifierData: FetchedData = {};
 
-    const resolvedFunction = async (rawDataSample: [string, BigNumberish]) => {
-      let isResolved = false;
-      for (const resolverObject of this.resolverRouter) {
-        if (
-          resolverObject.regExp &&
-          resolverObject.regExp.test(rawDataSample[0])
-        ) {
-          const resolvedAccount = await resolverObject.resolver.resolve(
-            rawDataSample[0]
-          );
-          if (resolvedAccount !== "undefined") {
-            if (!accountTypes.includes(resolverObject.accountType)) {
-              accountTypes.push(resolverObject.accountType);
-            }
+    const rawDataByAccountType: AccountsData = {};
 
-            resolvedIdentifierData[resolvedAccount] = rawDataSample[1];
-            isResolved = true;
+    let canBeResolved = false;
+    for (const [key, value] of Object.entries(rawData)) {
+      canBeResolved = false;
+      for (const resolverObject of this.resolverRouter) {
+        if (resolverObject.regExp && resolverObject.regExp.test(key)) {
+          canBeResolved = true;
+          if (!accountSources.includes(resolverObject.accountSource)) {
+            accountSources.push(resolverObject.accountSource);
+          }
+          if (
+            !Object.prototype.hasOwnProperty.call(
+              rawDataByAccountType,
+              resolverObject.accountType
+            )
+          ) {
+            rawDataByAccountType[resolverObject.accountType] = {};
```

Comment:
Shouldn't you declare accountSource instead? 
The `AccountsData` type does not have any accountType property

Reply:
You are tight, good catch!
I forgot to change the accountSource for accountType, but this cause no problems because it's an object of array (I only do for loop on it)